packages/clay-checkbox/src/ClayCheckbox.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
+	containerProps?: React.HTMLAttributes<HTMLDivElement>;
+	indeterminate?: boolean;
+	inline?: boolean;
+	label?: React.ReactElement | React.ReactText;
+}
+
+const ClayCheckbox: React.FunctionComponent<Props> = ({
+	checked = false,
+	className,
+	containerProps = {className: ''},
+	indeterminate = false,
+	inline,
+	label,
+	...otherProps
+}) => {
+	const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+	React.useEffect(() => {
+		if (inputRef.current) {
+			inputRef.current.indeterminate = indeterminate;
+		}
+	}, [indeterminate]);
+
+	return (
+		<div
+			{...containerProps}
+			className={classNames(
+				'custom-control custom-checkbox',
+				containerProps.className,
+				{
+					'custom-control-inline': inline,
+				}
+			)}
+		>
+			<label>
+				<input
+					{...otherProps}
+					checked={checked}
+					className={classNames('custom-control-input', className)}
+					ref={inputRef}
+					type="checkbox"
+					value={checked.toString()}
+				/>
+
+				{label && (
+					<span className="custom-control-label">
```

Comment:
This should be out of the condition, just your content, we should also add an API or use the children so we can add other content that is not text, we will be using it in a ClayCard for example.

See https://github.com/liferay/clay/blob/2.x-develop/packages/clay-checkbox/src/ClayCheckbox.soy#L93-L108

Reply:
I was trying to avoid using multiple inter-dependent props such as `showLabel`, `label`, and `children`. By using just label, we can assure its the only place to add content. One way to allow for more than text would be to check the type of the label and conditionally render the class names. What do you think?

Reply:
Hmm, just be concerned that the API name is not self explanatory for this use case. But it sounds good to me to.

Reply:
Ill add it right now and test it out. Maybe be some edge case where a user wants both text and markup. Ill update the pr and let you know